documentation/src/main/asciidoc/search-dsl.asciidoc

```diff
@@ -78,20 +78,107 @@ include::todo-placeholder.asciidoc[]
 
 include::todo-placeholder.asciidoc[]
 
-// TODO [[]]:
 // TODO [[_raise_an_exception_on_time_limit]] https://docs.jboss.org/hibernate/search/5.11/reference/en-US/html_single/#_raise_an_exception_on_time_limit
 // TODO [[_limit_the_number_of_results_when_the_time_limit_is_reached]] https://docs.jboss.org/hibernate/search/5.11/reference/en-US/html_single/#_limit_the_number_of_results_when_the_time_limit_is_reached
 
 [[search-dsl-query-entity-loading-options]]
 === Entity loading options
+
+Hibernate Search executes database queries to load entities
+that are returned as part of the hits of a search query.
+
+This section presents all available options related to entity loading in search queries.
+
+[[search-dsl-query-cache-lookup-strategy]]
+==== Cache lookup strategy
+// Search 5 anchors backward compatibility
+[[_customizing_object_initialization_strategies]]
+
+By default, Hibernate Search will load entities from the database directly,
+without looking at any cache.
+This is a good strategy when the size of caches (Hibernate ORM session or second level cache)
+is much lower than the total number of indexed entities.
+
+If a significant portion of your entities are present in the second level cache,
+you can force Hibernate Search to retrieve entities from the persistence context (the session)
+and/or the second level cache if possible.
+Hibernate Search will still need to execute a database query to retrieve entities missing from the cache,
+but the query will likely have to fetch fewer entities,
+leading to better performance and lower stress on your database.
+
+This is done through the cache lookup strategy,
+which can be configured by setting the configuration property `hibernate.search.query.loading.cache_lookup_strategy`:
+
+* `skip` (the default) will not perform any cache lookup.
```

Comment:
From what I've gathered from Search 5's documentation, it seems that at some point, second level cache lookups were very inefficient with some cache implementations. That seems really weird, but we'd have to check that, and if it's not the case anymore, then yes, it would make sense to use all caches by default.